### PR TITLE
Update dev workflow with boolean requirements for dev workflows

### PR DESCRIPTION
**Pull request by alexiswl**
## TODO
- [x] #257 

## Dragen alignment with qc pipeline
* added _enable_duplicate_marking_, _enable_map_align_output_ and _enable_sort_ to qc pipeline whilst removing _sites_somalier_ and _reference_fasta_ as input keys.

## Dragen somatic pipeline
* No changes required

## Dragen transcriptome pipeline
* added _enable_map_align_output_ and _enable_duplicate_marking_ (default false) as input keys 
* updated transcriptome reference directories to include quantification coverage (d309494)

## Related PRs
* umccr/cwl-ica#151
* umccr/cwl-ica#152

### CODE REVIEW

**Comment by alexiswl** [comment]
@skanwal feel free to add to this commit if you wish to update the reference samples for the wts workflow dev inputs, see https://github.com/umccr/cwl-ica/issues/120 for more info.

**Comment by skanwal** [comment]
Updated in WTS workflow inputs in https://github.com/umccr/infrastructure/pull/260/commits/d3094943dddfa9a353a8a69985c25565b3bc177d.
Note: I also need to update RNAsum workflow deets. It'll be helpful to keep this around until I have made changes for that workflow as well.

**Comment by victorskl** [comment]
Dear @alexiswl @skanwal -- Whenever this PR is ready/final, please kindly lift `hold merging` flag and, please mark it `Ready for review`. Thank you!

**Comment by alexiswl** [comment]
> Dear @alexiswl @skanwal -- Whenever this PR is ready/final, please kindly lift `hold merging` flag and, please mark it `Ready for review`. Thank you!

Thanks Victor, only one on my todo is the https://github.com/umccr/infrastructure/pull/257 PR, which this branch is branched off? If you'd rather wait since #257 also contains the production deployments, then this should be good to merge from my end. Will let @skanwal have the final say for the wts and rnasum workflow statuses in development

**Comment by victorskl** [comment]
Ah. 257. Of course. No worries. Thanks for reminder.

**Comment by skanwal** [comment]
All changes for WTS and RNAsum (dev) are now in.

**Comment by victorskl** [comment]
(Eh. I will put you back to assignee, Sehrish. It is a record of work done for the task. 😃)

Thanks for update too, Sehrish. Also. 257 is clear. I will lift the `hold merging` flag.

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 1]
```diff
@@ -124,14 +124,9 @@ locals {
       "output_file_prefix": null,
       "output_directory": null,
       "fastq_list_rows": null,
-      "sites_somalier": {
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:127]: Right! So this removed somalier step at workflow side now. Got it!

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 2]
```diff
@@ -124,14 +124,9 @@ locals {
...
-        "class": "File",
-        "location": "gds://development/reference-data/somalier/sites.hg38.rna.vcf.gz"
-      },
-      "reference_fasta": {
-        "class": "File",
-        "location": "gds://development/reference-data/genomes/hg38/hg38.fa"
-      },
+      "enable_duplicate_marking": true,
+      "enable_map_align_output": true,
+      "enable_sort": true,
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:129]: Right. So I understood that, these flags were there previously and, now make them explicit in workflow input template config to better capture these, correct?

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 3]
```diff
@@ -218,6 +213,8 @@ locals {
       "fastq_list_rows": null,
       "output_file_prefix": null,
       "output_directory": null,
+      "enable_map_align_output": true,
+      "enable_duplicate_marking": false,
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:217]: Right. That's where Pete mentioned, for transcriptome, this is false. Ok, gotcha. Also, understood that these will be static flag at input template (ditto above).

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 4]
```diff
@@ -245,23 +242,23 @@ locals {
...
             },
             {
                 "class": "Directory",
-                "location": "gds://development/reference-data/dragen_wts/multiqc/2022-08-17--3.9.3/SBJ01286"
+                "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01286"
             },
             {
                 "class": "Directory",
-                "location": "gds://development/reference-data/dragen_wts/multiqc/2022-08-17--3.9.3/SBJ01673"
+                "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01673"
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:261]: Double checked and I can confirm that these paths exist in GDS, yes. FYI.

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 5]
```diff
@@ -382,8 +379,8 @@ locals {
   }
 
   umccrise_wfl_version = {
-    dev = "2.1.1--3.9.3"
-    prod = "2.1.1--3.9.3--e4195a6"
+    dev = "2.2.0--3.9.3"
+    prod = "2.2.0--3.9.3--4e00721"
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:383]: This branch is worked on top of 257. So, this is ok. Just noting.

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 6]
```diff
@@ -394,6 +391,7 @@ locals {
       "output_directory_germline": null,
       "output_directory_umccrise": null,
       "output_file_prefix_germline": null,
+      "enable_duplicate_marking": true,
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:394]: ditto 257, so all good, just noting.

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 7]
```diff
@@ -412,6 +410,7 @@ locals {
...
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:413]: ditto 257, so all good, just noting.

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 8]
```diff
@@ -431,7 +430,7 @@ locals {
   }
 
   rnasum_wfl_version = {
-    dev = "0.4.4"
+    dev = "0.4.5"
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:433]: Yep. I can confirm `0.4.5` tag exists. All good. Just noting.

**Review by victorskl — APPROVED** [review]
LGTM! Comments were just notes; no action required. Confirming understanding. @reisingerf @skanwal over to you! 🙂

File: terraform/stacks/umccr_data_portal/workflow/main.tf (hunk 4 above)
Comment (victorskl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf:261]: Also accessible by Portal service account.

File: terraform/stacks/umccr_data_portal/workflow/main.tf (hunk 2 above)
Comment (alexiswl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf:129]: Yep that's correct

File: terraform/stacks/umccr_data_portal/workflow/main.tf (hunk 3 above)
Comment (alexiswl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf:217]: Yep

File: terraform/stacks/umccr_data_portal/workflow/main.tf (hunk 5 above)
Comment (alexiswl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf]: Have merged master back into this branch, so changes are now only relevant to this branch

File: terraform/stacks/umccr_data_portal/workflow/main.tf (hunk 7 above)
Comment (alexiswl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf]: Have merged master back into this branch, so changes are now only relevant to this branch

File: terraform/stacks/umccr_data_portal/workflow/main.tf (hunk 1 above)
Comment (skanwal) [reply terraform/stacks/umccr_data_portal/workflow/main.tf:127]: Which reminds me I need to do the same for transcriptome workflow as well. 
@alexiswl - should we submit the changes for that in this same PR or a new one? 
I am happy with both options.

Comment (alexiswl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf:127]: Might as well do it now, do you need to update the workflow as well. That shouldn't take too long.

Comment (skanwal) [reply terraform/stacks/umccr_data_portal/workflow/main.tf:127]: Yes, I am on it. Will ping you on PR in `cwl-ica` in a tick.

**Review by reisingerf — APPROVED** [review]
Yup, no issues from my side

**Review by victorskl — COMMENTED** [review]
Thanks all. Can we merge this? Bec it is dependency PR of #261

**Comment by victorskl** [comment]
Dear @skanwal if you find this ready and tick approve, I will merge it. Thanks

**Review by skanwal — APPROVED** [review]
LGTM. Thanks @victorskl

**Comment by victorskl** [comment]
Merging...

**Pull request merged**